design/sketch.webidl

```diff
@@ -0,0 +1,532 @@
+typedef long u32;
+
+// ****************************************************************************
+// SHADER RESOURCES (buffer, textures, texture views, samples)
+// ****************************************************************************
+
+// Buffer
+typedef u32 WebGPUBufferUsageFlags;
+interface WebGPUBufferUsage {
+    const u32 NONE = 0;
+    const u32 MAP_READ = 1;
+    const u32 MAP_WRITE = 2;
+    const u32 TRANSFER_SRC = 4;
+    const u32 TRANSFER_DST = 8;
+    const u32 INDEX = 16;
+    const u32 VERTEX = 32;
+    const u32 UNIFORM = 64;
+    const u32 STORAGE = 128;
+};
+
+dictionary WebGPUBufferDescriptor {
+    u32 size;
+    WebGPUBufferUsageFlags usage;
+};
+
+interface WebGPUBuffer {
+    void setSubData(ArrayBuffer data, u32 offset);
+    Promise<ArrayBuffer> readbackAsync(u32 offset, u32 size);
+};
+
+// Texture view
+dictionary WebGPUTextureViewDescriptor {
+    // TODO Investigate what goes in there.
+};
+
+interface WebGPUTextureView {
+};
+
+// Texture
+typedef u32 WebGPUTextureDimensionEnum;
+interface WebGPUTextureDimension {
+    const u32 e1D = 0;
```

Comment:
`TEXTURE_nD`? It'd be a bit strange to have Hungarian notation in one place:)

Reply:
Yeah it's weird. Let's defer this I'm sure we'll have a bunch of discussions about naming.